diff --git a/CMakeLists.txt b/CMakeLists.txt
index fd1d7c7..777a05e 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -25,18 +25,35 @@ find_path(GSL_INCLUDE_DIR NAMES gsl/gsl_version.h)
 find_package(Threads REQUIRED)
 find_package(LibXml2 REQUIRED)
 find_package(PNG REQUIRED)
+find_package(Python COMPONENTS Interpreter Development)
+if(NOT Python_FOUND)
+  message(STATUS "Falling back to deprecated Python search.")
+  find_package(PythonInterp REQUIRED)
+  find_package(PythonLibs REQUIRED)
+  set(Python_VERSION ${PYTHON_VERSION_STRING})
+  set(Python_VERSION_MAJOR ${PYTHON_VERSION_MAJOR})
+  set(Python_VERSION_MINOR ${PYTHON_VERSION_MINOR})
+  set(Python_LIBRARIES ${PYTHON_LIBRARIES})
+  set(Python_INCLUDE_DIRS ${PYTHON_INCLUDE_DIRS})
+endif()
+message(STATUS "Using python version ${Python_VERSION}")
 #Prevent accidentally finding old BoostConfig.cmake file from casapy
 set(Boost_NO_BOOST_CMAKE ON)
-#set(PY_VERSION 3)
-#find_package(PythonLibs ${PY_VERSION} REQUIRED)
-find_package(PythonInterp REQUIRED)
-#find_package(PythonLibs ${PYTHON_VERSION_STRING} REQUIRED)
-find_package(PythonLibs 2.7 REQUIRED)
-#message(STATUS "Using python version ${PYTHON_VERSION_STRING}")
 # At least the intrusive_ref_counter needs Boost 1.55 (because the header was moved)
-find_package(Boost 1.55.0 REQUIRED COMPONENTS date_time filesystem python system)
-#find_package(Boost REQUIRED COMPONENTS date_time filesystem python-py36 system)
+# Bumped to 1.67 to be able to use proper python suffix. One attempt
+# to support at least two boost_python naming schemes here.
+set(boostpython python${Python_VERSION_MAJOR}${Python_VERSION_MINOR})
+find_package(Boost 1.67.0 COMPONENTS date_time filesystem ${boostpython} system)
+if(NOT Boost_FOUND)
+  message(STATUS "Falling back to old Boost search.")
+  set(boostpython python)
+  find_package(Boost 1.55.0  REQUIRED COMPONENTS date_time filesystem ${boostpython} system)
+endif()
+
 find_library(FFTW3_LIB fftw3 REQUIRED)
+# Not sure if this is needed everywhere, but I get linker
+# errors with missing symbols if not explicitly linking to fftw3_threads.
+find_library(FFTW3_THREADS_LIB fftw3_threads OPTIONAL)
 enable_language(Fortran OPTIONAL)
 find_package(BLAS REQUIRED)
 find_package(LAPACK REQUIRED)
@@ -53,7 +70,7 @@ include_directories(${FFTW3_INCLUDE_DIR})
 if(GSL_INCLUDE_DIR)
 	include_directories(${GSL_INCLUDE_DIR})
 endif(GSL_INCLUDE_DIR)
-include_directories(${PYTHON_INCLUDE_DIRS})
+include_directories(${Python_INCLUDE_DIRS})
 
 # The following stuff will set the "rpath" correctly, so that
 # LD_LIBRARY_PATH doesn't have to be set.
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 4a36e1c..55713b2 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -235,17 +235,17 @@ set(PYTHON_FILES
 
 set(ALL_LIBRARIES
 	${BLAS_LIBRARIES} ${LAPACK_LIBRARIES}
-	${Boost_SYSTEM_LIBRARY}
-	${Boost_FILESYSTEM_LIBRARY} ${Boost_DATE_TIME_LIBRARY}
-#	${Boost_PYTHON-PY36_LIBRARIES}
-	${Boost_PYTHON_LIBRARIES}
-	${FFTW3_LIB}
+	Boost::system
+	Boost::filesystem
+	Boost::date_time
+	Boost::${boostpython}
+	${FFTW3_THREADS_LIB} ${FFTW3_LIB}
 	${CASACORE_LIBRARIES}
 	${LAPACK_lapack_LIBRARY}
 	${CFITSIO_LIBRARY}
 	${CMAKE_THREAD_LIBS_INIT}
 	${PNG_LIBRARIES} ${LIBXML2_LIBRARIES}
-	${PYTHON_LIBRARIES})
+	${Python_LIBRARIES})
 
 if(GTKMM_FOUND)
 	message(STATUS "GTKMM found.") 
